languages/python/reference/exercise-concepts/markdown.md

```diff
@@ -0,0 +1,48 @@
+
+- `import` key word/ importing:  Both the original code to be refactored for the exercise and the example solution use the `import` keyword to import the `re` module in support of Regular Expressions in python.
+
+- `str.format()` :  The example solution uses `str.split()` to break the passed in markdown string into a list of lines broken up by the `\n` character.
+
+- `re.match()`:  Method of the `re` class that returns a `match` object with any matched values from a specified Regualr Expression or pre-compliled Regular Expression.  The example uses `re.match()` in multiple places to search for text patterns that need re-formatting or subsitituting. 
+
+- regular expressions:  As a seperate language with a seperate interpreter that is often embeded inside another programming language.  Like many other programming lanagues in use, python supports the Regular Expression language and implements a Regular Expression interpreter in the core `re` module.  Both the original code to be refactored for this exercise and the example solution use the core `re` module to access Regualr Expression functionality.
+
+- `return` keyword:  Most of the functions in the example solution specify a _return_ value using the `return` keyword.
+
+- `None`:  None (null) type.  Used to define a null variable or an object.  It is in of itself a singleton in any given python program.  The example uses `None` as an expecit return value in several functions.
+
+- booleans:  True and False of type `bopl`.  The example solution uses `True` and `False` as return values from functions that test membership in a list of values.
+
+- assignment:  The example solution uses assingment for variables and other values.
+
+- `re.sub()`:  Metho of the `re` class that replaces a Regular Expression match with a new value.  The example solution uses this method in various places to susititue _markdown_ syntax for _HTML_ syntax in the passed-in markdown text.
+
+- dictionaries:  Mapping type.  The example solution employes a dictionary to return values from the `parse_line()` function.
+
+- `str.split()`:  String method.  The example solution uses `str.split()` to split the incoming string into lines (_based on `\n`_), so that the text can be more easily processed for output.
+
+- for loops:  The example solution uses `for` loops to iterate over various function inputs.
+
+- iteration/iterable:  The example solution uses the `for _ in _` syntax to iterate over a list of lines.  This is possible because a list is an `iterable`.
+
+- conditionals:  The example solution uses `if` to check for pattern matching and membershihp conditions in differnt functions for processing different markdown patterns.
+
+- `match.group`:  A retun object of the `re.match()` method.  Returned when there are one or more Regular Expression groups specified in the RE pattern.
```

Comment:
Spelling: "retun" -> "return". This is a bit confusing; the **re.match** _function_ returns an **re.Match** instance, and that instance in turn has the **Match.group** method, but that method exists even if there are no groups specified in the pattern.

Reply:
- `match.group`:  A return object of the `re.match()` function. Returns a `re.Match` _instance_ which in turn has a `Match.group` method. `Match.group` exists even if there are no groups specified in the pattern.  See [Match.group](https://docs.python.org/3/library/re.html#re.Match.group) in the standard docs for more detail.